### PR TITLE
Task/mes 3089 display rekey reason card

### PR DESCRIPTION
**Pull request by mgoldson**
## Description and relevant Jira numbers
Display a new card with Rekey reason on the completed tests search results screen.

**Related PR's**
- https://github.com/dvsa/mes-mobile-app/pull/743

Opted to go with a task -> feature branch strategy for this piece of work whilst building understanding of the codebase. Let me know if you prefer to go straight to develop with this.

https://jira.dvsacloud.uk/browse/MES-3089

## Pull Request checklist

- [x] [WIP] tag removed from PR title
- [x] PR has an understandable description

## Git feature branch checklist:

- [x] branch name comply with our branching strategy
- [x] git branch contains relevant JIRA ticket number
- [x] branch rebased against the latest develop
- [ ] commits are squashed

## Sign off process checklist:

- [x] Code has been tested manually
- [x] Tests are passing
- [x] PR link added to JIRA
- [x] Reviewers selected in Github
- [ ] Any new reusable component/widget added to component library on Confluence

- [ ] Screenshot(s) captured on the physical device (if applicable)
- [ ] Tested by QA
- [ ] PO's approval

Simulator Screen Shot - iPad Pro (10 5-inch) - 2019-09-09 at 17 31 32

### CODE REVIEW

File: src/pages/view-test-result/view-test-result.ts [hunk 1]
```diff
@@ -333,6 +335,58 @@ export class ViewTestResultPage extends BasePageComponent implements OnInit {
...
+      const ipadIssueText = {
+        broken: 'Broken',
+        lost: 'Lost',
+        technicalFault: 'Technical fault',
+        stolen: 'Stolen',
+      };
+
+      let value = '';
+
+      for (const prop in reasonType) {
```
Comment (szabikr) [inline src/pages/view-test-result/view-test-result.ts]: Don't think that we need a loop here as we just want to return a string considering the properties on reasonType and we know that there are only 4 of them. Usage of if statements would be simple and clean.

File: src/pages/view-test-result/components/rekey-details-card/rekey-details.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+import { RekeyDetailsModel } from './rekey-details-card.model';
+
+@Component({
+  selector: 'rekey-details-card',
+  templateUrl: 'rekey-details-card.html',
+})
+export class RekeyDetailsCardComponent {
+
+  @Input()
+  data: RekeyDetailsModel;
```
Comment (szabikr) [inline src/pages/view-test-result/components/rekey-details-card/rekey-details.ts]: `data` is very generic name. We could use a more verbose one like `rekeyDetails`

File: src/pages/view-test-result/components/rekey-reason-card/rekey-reason.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+import { RekeyReason } from '@dvsa/mes-test-schema/categories/B';
+
+@Component({
+  selector: 'rekey-reason-card',
+  templateUrl: 'rekey-reason-card.html',
+})
+export class RekeyReasonCardComponent {
+
+  @Input()
+  data: RekeyReason;
```
Comment (szabikr) [inline src/pages/view-test-result/components/rekey-reason-card/rekey-reason.ts]: Same here, `rekeyReason` would be a more suitable name

**Review by szabikr — APPROVED** [review]
Great stuff, I like your branching strategy! Added a few comments, they are not show stoppers

**Review by ammarhaiderbjss — APPROVED** [review]

File: src/pages/view-test-result/components/rekey-details-card/rekey-details.ts (hunk 2 above)
Comment (mgoldson) [reply src/pages/view-test-result/components/rekey-details-card/rekey-details.ts]: I agree. However, this follows the convention on all of the other components within the view-test-result page. Happy to change it but are we comfortable changing it for the rekey components only and leaving the others as is?

File: src/pages/view-test-result/view-test-result.ts (hunk 1 above)
Comment (mgoldson) [reply src/pages/view-test-result/view-test-result.ts]: Ok, happy to change to if's if you think that's cleaner.

**Review by MarkPatt89 — APPROVED** [review]

**Review by rossfauldsbjss — APPROVED** [review]
looks good

**Comment by mgoldson** [comment]
@szabolcs-keresztes Removed the for loop and changed the generic data property.

**Pull request merged**